### PR TITLE
Convert HTML to Markdown on the code generation page

### PR DESCRIPTION
**Pull request by leviv**
## This PR is a:

- [ ] New topic
- [ ] Content update
- [ ] Content fix or rewrite
- [x] Bug fix or improvement

## Summary
Convert HTML to Markdown on the code generation page

When this pull request is merged, it will...
Make the source code easier to read

Converts HTML tags to their markdown equivalent on the code generation page

### CODE REVIEW

File: guides/v2.1/extension-dev-guide/code-generation.md [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
-menu_title: Code generation
-menu_order: 8
 version: 2.1
 github_link: extension-dev-guide/code-generation.md
 ---
 
-<h2 id="codegen-over">Overview of code generation</h2>
-The Magento application generates code to create non-existent classes. As an example, look at the <a href="{{ site.mage2000url }}/app/code/Magento/Customer/Model/ResourceModel/AddressRepository.php" target="_blank">\Magento\Customer\Model\Resource\AddressRepository</a> constructor. A snippet follows:
+## Overview of code generation {#codegen-over}
+The Magento application generates code to create non-existent classes. As an example, look at the [\\Magento\\Customer\\Model\\Resource\\AddressRepository] constructor. A snippet follows:
```
Comment (shrielenee) [inline guides/v2.1/extension-dev-guide/code-generation.md]: Why are there two \\ ? Should be [Magento/Customer/Model/Resource/AddressRepository]

File: guides/v2.1/extension-dev-guide/code-generation.md [hunk 2]
```diff
@@ -47,9 +44,16 @@ Code generation is required in Magento 2. Generating code assures you of the fol
...
-
-<h2 id="codegen-om">Object Manager responsibility for code compilation</h2>
+## Object Manager responsibility for code compilation {#codegen-om}
 When code changes as discussed in the preceding section, the Object Manager compiles it.
 
 
 The code compiler compiler creates `var/di/global.ser`, which is a PHP serialized map of all constructor definitions mixed with object linking configuration defined in di.xml. `di.xml` is the dependency injection configuration. There is a global `app/etc/di.xml` and there can one defined for every {% glossarytooltip c1e4242b-1f1a-44c3-9d72-1d5b1435e142 %}module{% endglossarytooltip %}.
+
+<!-- Link Definitions -->
+[\\Magento\\Customer\\Model\\Resource\\AddressRepository]: {{ site.mage2000url }}/app/code/Magento/Customer/Model/ResourceModel/AddressRepository.php
```
Comment (shrielenee) [inline guides/v2.1/extension-dev-guide/code-generation.md]: Make sure the slashes in the directory are correct.

File: guides/v2.1/extension-dev-guide/code-generation.md [hunk 3]
```diff
@@ -47,9 +44,16 @@ Code generation is required in Magento 2. Generating code assures you of the fol
...
+{target="_blank"}
```
Comment (shrielenee) [inline guides/v2.1/extension-dev-guide/code-generation.md]: Should be {:target="_blank"}

File: guides/v2.2/extension-dev-guide/code-generation.md [hunk 4]
```diff
@@ -1,15 +1,12 @@
...
-menu_title: Code generation
-menu_order: 8
 version: 2.2
 github_link: extension-dev-guide/code-generation.md
 ---
 
-<h2 id="codegen-over">Overview of code generation</h2>
-The Magento application generates code to create non-existent classes. As an example, look at the <a href="{{ site.mage2000url }}/app/code/Magento/Customer/Model/ResourceModel/AddressRepository.php" target="_blank">\Magento\Customer\Model\Resource\AddressRepository</a> constructor. A snippet follows:
+## Overview of code generation {#codegen-over}
+The Magento application generates code to create non-existent classes. As an example, look at the [\\Magento\\Customer\\Model\\Resource\\AddressRepository] constructor. A snippet follows:
```
Comment (shrielenee) [inline guides/v2.2/extension-dev-guide/code-generation.md]: See previous comment about slashes.

File: guides/v2.2/extension-dev-guide/code-generation.md [hunk 5]
```diff
@@ -48,8 +45,15 @@ Code generation is required in Magento 2. Generating code assures you of the fol
...
 
-<h2 id="codegen-om">Object Manager responsibility for code compilation</h2>
+## Object Manager responsibility for code compilation {#codegen-om}
 When code changes as discussed in the preceding section, the Object Manager compiles it.
 
-
 The code compiler creates `generated/metadata/global.php`, which is a PHP serialized map of all constructor definitions mixed with object linking configuration defined in di.xml. `di.xml` is the dependency injection configuration. There is a global `app/etc/di.xml` and there can one defined for every {% glossarytooltip c1e4242b-1f1a-44c3-9d72-1d5b1435e142 %}module{% endglossarytooltip %}.
+
+<!-- Link Definitions -->
+[\\Magento\\Customer\\Model\\Resource\\AddressRepository]: {{ site.mage2000url }}/app/code/Magento/Customer/Model/ResourceModel/AddressRepository.php
```
Comment (shrielenee) [inline guides/v2.2/extension-dev-guide/code-generation.md]: Slashes and target code formatting - see previous comments.

File: guides/v2.1/extension-dev-guide/code-generation.md (hunk 1 above)
Comment (leviv) [reply guides/v2.1/extension-dev-guide/code-generation.md]: Fixed

File: guides/v2.1/extension-dev-guide/code-generation.md (hunk 2 above)
Comment (leviv) [reply guides/v2.1/extension-dev-guide/code-generation.md]: Fixed

File: guides/v2.1/extension-dev-guide/code-generation.md (hunk 3 above)
Comment (leviv) [reply guides/v2.1/extension-dev-guide/code-generation.md]: Fixed

File: guides/v2.2/extension-dev-guide/code-generation.md (hunk 4 above)
Comment (leviv) [reply guides/v2.2/extension-dev-guide/code-generation.md]: Fixed

File: guides/v2.2/extension-dev-guide/code-generation.md (hunk 5 above)
Comment (leviv) [reply guides/v2.2/extension-dev-guide/code-generation.md]: Fixed

**Comment by leviv** [comment]
@shrielenee Made necessary changes

**Comment by shrielenee** [comment]
Thanks @leviv for making those changes! Good work.

**Pull request merged**